### PR TITLE
Add tuple declarations to AllInOneCSharpCode

### LINKED ISSUE #12446: TestResource.AllInOneCSharpCode is missing VariableDeconstructionDeclarator
Our test resource that has an example of every syntax node is missing `VariableDeconstructionDeclarator`.

### PR DESCRIPTION
**Pull request by khyperia**
Also modify the test that checks for all SyntaxKinds to be present to
fail if an "expected to be missing" SyntaxKind is actually present.

Fixes #12446

Test-only change.

### CODE REVIEW

**Comment by khyperia** [comment]
retest windows_release_vs-integration_prtest please

Probably xunit crashed.

**Comment by khyperia** [comment]
retest windows_release_vs-integration_prtest please

**Comment by khyperia** [comment]
Ping @dotnet/roslyn-compiler for review.

File: src/Compilers/CSharp/Test/Semantic/Diagnostics/DiagnosticAnalyzerTests.AllInOne.cs [hunk 1]
```diff
@@ -25,20 +25,11 @@ public void DiagnosticAnalyzerAllInOne()
...
-            syntaxKindsMissing.Add(SyntaxKind.ForEachVariableStatement);
-            syntaxKindsMissing.Add(SyntaxKind.DeclarationExpression);
-            syntaxKindsMissing.Add(SyntaxKind.DiscardDesignation);
-
             var analyzer = new CSharpTrackingDiagnosticAnalyzer();
             CreateCompilationWithMscorlib45(source).VerifyAnalyzerDiagnostics(new[] { analyzer });
             analyzer.VerifyAllAnalyzerMembersWereCalled();
             analyzer.VerifyAnalyzeSymbolCalledForAllSymbolKinds();
-            analyzer.VerifyAnalyzeNodeCalledForAllSyntaxKinds(syntaxKindsMissing);
+            analyzer.VerifyAnalyzeNodeCalledForAllSyntaxKinds(new HashSet<SyntaxKind>());
```
Comment (jaredpar) [inline src/Compilers/CSharp/Test/Semantic/Diagnostics/DiagnosticAnalyzerTests.AllInOne.cs]: Much better. Fixed forever 😄

File: src/Test/Utilities/Portable/Diagnostics/TrackingDiagnosticAnalyzer.cs [hunk 2]
```diff
@@ -92,7 +92,11 @@ protected virtual bool IsAnalyzeNodeSupported(TLanguageKindEnum syntaxKind)
         public void VerifyAnalyzeNodeCalledForAllSyntaxKinds(HashSet<TLanguageKindEnum> syntaxKindsPatterns)
         {
             var expectedSyntaxKinds = AllSyntaxKinds.Where(IsAnalyzeNodeSupported);
-            var actualSyntaxKinds = _callLog.Where(a => FilterByAbstractName(a, "SyntaxNode")).Select(e => e.SyntaxKind).Concat(syntaxKindsPatterns).Distinct();
+            var actualSyntaxKinds = new HashSet<TLanguageKindEnum>(_callLog.Where(a => FilterByAbstractName(a, "SyntaxNode")).Select(e => e.SyntaxKind));
+            var savedSyntaxKindsPatterns = new HashSet<TLanguageKindEnum>(syntaxKindsPatterns);
+            syntaxKindsPatterns.IntersectWith(actualSyntaxKinds);
```
Comment (jcouv) [inline src/Test/Utilities/Portable/Diagnostics/TrackingDiagnosticAnalyzer.cs]: Consider renaming `syntaxKindsPatterns` to `expectedSyntaxKinds`.

File: src/Test/Utilities/Portable/TestResource.resx [hunk 3]
```diff
@@ -300,6 +300,7 @@ namespace My
                 0f,
                 1.1f
             };
+            var (tuple, _) = (named: 1, 2);
```
Comment (jcouv) [inline src/Test/Utilities/Portable/TestResource.resx]: Is there a similar AllInOne test for VB?

Comment (khyperia) [inline src/Test/Utilities/Portable/TestResource.resx]: Yes. The test here https://github.com/dotnet/roslyn/blob/master/src/Compilers/VisualBasic/Test/Semantic/Diagnostics/DiagnosticAnalyzerTests.AllInOne.vb#L23 calls the same method as the C# one, and already has an empty HashSet. However, I just noticed this line https://github.com/dotnet/roslyn/blob/master/src/Test/Utilities/Portable/Diagnostics/TrackingDiagnosticAnalyzer.cs#L63 which apparently has `TupleType|TupleElement|TupleExpression.*` and so was ignoring the lack of tuples in the VB test, as I don't think VB has anything named like `SyntaxKind.SingleVariableDesignation` and friends, only `SyntaxKind.TupleExpression` and friends.

I'll push a new commit addressing that soon.

**Comment by khyperia** [comment]
retest windows_release_vs-integration_prtest please

**Comment by khyperia** [comment]
Justification for retest: `windows_release_vs-integration_prtest` test `Roslyn.VisualStudio.IntegrationTests.VisualBasic.BasicExpressionEvaluator.EvaluateInvalidExpressions` failed with `System.Runtime.InteropServices.COMException : Operation not supported. Unknown error: 0x8971003c.`, pretty sure that's flakiness and not this change.

File: src/Test/Utilities/Portable/TestResource.resx [hunk 4]
```diff
@@ -961,7 +961,7 @@ L:
             Dim o4 = New MyObject(dynamic) With {.A = 0, .B = 0, .C = 0}
             Dim o5 = New With {Key .A = 0}
             Dim a() As Integer = {0, 1, 2, 3, 4, 5}
-            Dim a As (named As Integer, Integer) = (named:=1, 2)
+            Dim a As(named As Integer, Integer) =(named:=1, 2)
```
Comment (jcouv) [inline src/Test/Utilities/Portable/TestResource.resx]: Do you have a follow-up issue to sort this out? As we discussed yesterday, you shouldn't need this workaround for this test to pass.

Comment (jcouv) [inline src/Test/Utilities/Portable/TestResource.resx]: Filed https://github.com/dotnet/roslyn/issues/20896

**Pull request merged**